### ISSUE TITLE
ParseTransactionError: UNIQUE constraint failed

### ISSUE THREAD

**Issue by droplister** [CounterpartyXCP/counterparty-lib#1144]
```
counterparty_1          | [2022-01-12 20:08:50][WARNING] ParseTransactionError for tx b6e15eb3b15073ac6de8996311a40eb2b47c80451494840b557e10bf0924d1c1: ConstraintError: UNIQUE constraint failed: assets.asset_id
counterparty_1          | [2022-01-12 20:08:50][ERROR] Unhandled Exception
counterparty_1          | Traceback (most recent call last):
counterparty_1          |   File "/counterparty-lib/counterpartylib/lib/blocks.py", line 108, in parse_tx
counterparty_1          |     issuance.parse(db, tx, message, message_type_id)
counterparty_1          |   File "/counterparty-lib/counterpartylib/lib/messages/issuance.py", line 440, in parse
counterparty_1          |     issuance_parse_cursor.execute(sql, bindings)
counterparty_1          |   File "src/cursor.c", line 236, in resetcursor
counterparty_1          | apsw.ConstraintError: ConstraintError: UNIQUE constraint failed: assets.asset_id
counterparty_1          |
counterparty_1          | During handling of the above exception, another exception occurred:
counterparty_1          |
counterparty_1          | Traceback (most recent call last):
counterparty_1          |   File "/usr/local/bin/counterparty-server", line 11, in <module>
counterparty_1          |     load_entry_point('counterparty-cli', 'console_scripts', 'counterparty-server')()
counterparty_1          |   File "/counterparty-cli/counterpartycli/__init__.py", line 16, in server_main
counterparty_1          |     server.main()
counterparty_1          |   File "/counterparty-cli/counterpartycli/server.py", line 166, in main
counterparty_1          |     server.start_all(db)
counterparty_1          |   File "/counterparty-lib/counterpartylib/server.py", line 489, in start_all
counterparty_1          |     blocks.follow(db)
counterparty_1          |   File "/counterparty-lib/counterpartylib/lib/blocks.py", line 1414, in follow
counterparty_1          |     new_ledger_hash, new_txlist_hash, new_messages_hash, found_messages_hash = parse_block(db, block_index, block_time)
counterparty_1          |   File "/counterparty-lib/counterpartylib/lib/blocks.py", line 207, in parse_block
counterparty_1          |     raise e
counterparty_1          |   File "/counterparty-lib/counterpartylib/lib/blocks.py", line 201, in parse_block
counterparty_1          |     parse_tx(db, tx)
counterparty_1          |   File "/counterparty-lib/counterpartylib/lib/blocks.py", line 147, in parse_tx
counterparty_1          |     raise exceptions.ParseTransactionError("%s" % e)
counterparty_1          | counterpartylib.lib.exceptions.ParseTransactionError: ConstraintError: UNIQUE constraint failed: assets.asset_id
federatednode_counterparty_1 exited with code 0

```

**Comment by droplister** [comment]
I think this could be related to:
https://github.com/CounterpartyXCP/counterparty-lib/blob/a99ddefd1a37751c9a090de63b3c106b82cb0bad/counterpartylib/lib/messages/issuance.py#L397

Because it's defined, but no check is in place between the definition and this evaluation:
https://github.com/CounterpartyXCP/counterparty-lib/blob/a99ddefd1a37751c9a090de63b3c106b82cb0bad/counterpartylib/lib/messages/issuance.py#L431

**Comment by droplister** [comment]
Fixed by #1145
